### PR TITLE
usm: Fixed races during components cleanups

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

wrong order of closing components, alongside counting of parallel cleanups can create errors log which can confuse us during escalations.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/shared_libraries.go [hunk 1]
```diff
@@ -207,14 +216,17 @@ func (w *soWatcher) Start() {
 		return
 	}
 
+	w.wg.Add(1)
 	go func() {
+		defer w.wg.Done()
 		defer cleanupExit()
-		defer w.processMonitor.Stop()
```
Comment (nplanel) [inline pkg/network/usm/shared_libraries.go:212]: why not calling stop anymore ?

**Review by nplanel — CHANGES_REQUESTED** [review]

File: pkg/network/usm/shared_libraries.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/usm/shared_libraries.go:212]: good catch, that will be removed in a future PR (the process monitor refactor)

File: pkg/network/usm/ebpf_ssl.go [hunk 2]
```diff
@@ -318,6 +317,7 @@ func (o *sslProgram) Start() {
 
 func (o *sslProgram) Stop() {
 	o.perfHandler.Stop()
+	o.watcher.Stop()
```
Comment (nplanel) [inline pkg/network/usm/ebpf_ssl.go:320]: o.watcher could be nil if we don't Start()

**Review by nplanel — APPROVED** [review]

File: pkg/network/usm/ebpf_ssl.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/usm/ebpf_ssl.go:320]: the only scenario in which we do not call `.Start` is if we shouldn't capture tls traffic, in such a scenario we don't call .Stop method as well

Comment (guyarb) [reply pkg/network/usm/ebpf_ssl.go:320]: Adding the program to `subprogram` array:
https://github.com/DataDog/datadog-agent/blob/b87a657c4a89383e7659c8b800ec5b30746643aa/pkg/network/usm/ebpf_main.go#L157-L161


Calling start and stop:
https://github.com/DataDog/datadog-agent/blob/b87a657c4a89383e7659c8b800ec5b30746643aa/pkg/network/usm/ebpf_main.go#L255-L277

Comment (nplanel) [reply pkg/network/usm/ebpf_ssl.go:320]: it's true today, but what about tomorrow, the code path would change and may segv

Comment (guyarb) [reply pkg/network/usm/ebpf_ssl.go:320]: that's true, but regardless of my change we'll have to change many methods due to that, so I'm postponning that for the time we do (if we do)

**Pull request merged**